Dockerfile.linux-build

```diff
@@ -0,0 +1,66 @@
+FROM ubuntu:18.04
+#
+# this file mirrors the build params used in the GitHub Actions and enables
+# reproducible builds for downstream forks and Ziti contributors 
+#
+
+ARG go_distribution_file=go1.16.9.linux-amd64.tar.gz
```

Comment:
is there a reason you couldn't pull from a particular go base image? also - ziti updates the version of go aggressively. i think 17 is already required. is it possible to pass this via variable?

Reply:
or maybe even use 'latest'? this feels like it could be a bit of a maintenance issue that will get lost. right now i think we rely on it being in the action's file but maybe we need to make this a variable for the project...

Reply:
after looking through the rest of the PR i'm wondering if this is this still needed or if it was solved on a different pr?

Reply:
It's still needed unless there's already a local reproducible build counterpart to the procedures that GitHub Actions uses to build for Linux. I didn't find one in the local development resources.

I'll send an update to track the version of Go that's actually needed, if possible.

Reply:
Also, would `/quickstart/local` be a better home for these?

Reply:
@plorenz Where is the recommended version of Go detectable inside the repo? Is there a `ziti-ci` command for that? I looked through the `ziti-ci` and `ziti` repos but didn't find it.

Reply:
You can look at the go.mod file for the ziti project. It has the required major release of go listed `go 1.17`. So currently the recommended version would be the latest patch version of 1.17.